app/src/main/java/com/example/ithaca_transit_android_v2/NetworkingAdapters.kt

```diff
@@ -44,4 +46,128 @@ class LocationAdapter {
         }
         return DataLocation(final)
     }
+}
+
+@JsonClass(generateAdapter = true)
+class RouteAdapter {
+    class DataRoute(
+        val data: DataRouteOptions
+    )
+
+    @JsonClass(generateAdapter = true)
+    class DataRouteOptions(
+        val fromStop: List<JsonRoute>,
+        val walking: List<JsonRoute>,
+        val boardingSoon: List<JsonRoute>
+    )
+
+    @JsonClass(generateAdapter = true)
+    class JsonRoute(
+        @Json(name = "directions")
+        val directions: List<JsonDirection>,
+        @Json(name = "startCoords")
+        val startLocation: JsonLocation,
+        @Json(name = "endCoords")
+        val endLocation: JsonLocation,
+        // TODO: need to calculate isWalkingOnly
+        @Json(name = "arrivalTime")
+        val arrival: String, // e.g. 2019-11-06T17:07:20Z
+        @Json(name = "departureTime")
+        val depart: String
+        // TODO: need to calculate BoardInMin
+    )
+
+    @JsonClass(generateAdapter = true)
+    class JsonDirection(
+        @Json(name = "path")
+        val listOfCoordinates: List<Coordinate>,
+        @Json(name = "startTime")
+        val startTime: String,
+        @Json(name = "endTime")
```

Comment:
We might need change our Direction model to also include distance, which is something that backend passes to us here.

 In our routeCards, we will probably need to display the time of each direction on the side.

Reply:
designs are kind of weird...we only need user distance from the location for the first set of directions? not sure how we want to translate that into our models? 

![image](https://user-images.githubusercontent.com/13595933/68907549-3b130a80-0716-11ea-9aca-60688b309044.png)

Reply:
Perhaps we can update the user's location everytime the user opens the app and then store that value as they search for destinations. It can be calculated from the busStop that backend returns using some Google API.